# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/10 12:27:49-08:00 okir@suse.de 
#   [NET]: Fix CMSG_COMPAT_OK length check.
#   
#   Need to check against struct compat_cmsghdr
#   not struct cmsghdr.
#   
#   Signed-off-by: Olaf Kirch <okir@suse.de>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# net/compat.c
#   2005/01/10 12:27:28-08:00 okir@suse.de +1 -1
#   [NET]: Fix CMSG_COMPAT_OK length check.
#   
#   Need to check against struct compat_cmsghdr
#   not struct cmsghdr.
#   
#   Signed-off-by: Olaf Kirch <okir@suse.de>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
diff -Nru a/net/compat.c b/net/compat.c
--- a/net/compat.c	2005-02-14 00:26:02 -08:00
+++ b/net/compat.c	2005-02-14 00:26:02 -08:00
@@ -125,7 +125,7 @@
 	 (struct compat_cmsghdr __user *)NULL)
 
 #define CMSG_COMPAT_OK(ucmlen, ucmsg, mhdr) \
-	((ucmlen) >= sizeof(struct cmsghdr) && \
+	((ucmlen) >= sizeof(struct compat_cmsghdr) && \
 	 (ucmlen) <= (unsigned long) \
 	 ((mhdr)->msg_controllen - \
 	  ((char *)(ucmsg) - (char *)(mhdr)->msg_control)))
